Orange/widgets/unsupervised/owhierarchicalclustering.py

```diff
@@ -1082,6 +1090,8 @@ def _update_labels(self):
         self.labels.setMinimumWidth(1 if labels else -1)
 
     def _invalidate_clustering(self):
+        if self.matrix is not None and not np.all(np.isfinite(self.matrix)):
+            return
```

Comment:
First, this check may be redundant since `np.all(np.isfinite(self.matrix))` is already checked above and `self.matrix` is set to None in this case. Hence, the situation in which this conjunction is true cannot arise, I guess.

Second -- if this condition can be fulfilled -- is skipping these calls really OK?

- `_update()` clears the plot and then performs further cleanup if `self.matrix is None`. I guess this must be done if the widget receives invalid data.
- `_update_labels()` checks for `self.root ` (which is set to `None` in `_update()`) and works correctly when `matrix` is `None`.
- `_invalidate_output()` has to be called pass `None` to outputs.

I believe that these two lines have to be removed. But please check whether there are any other side-effects.